File: src/main/java/lotto/utils/RandomGenerator.java
package lotto.utils;

import java.util.List;

import static camp.nextstep.edu.missionutils.Randoms.pickUniqueNumbersInRange;

public final class RandomGenerator {

    private static final int MIN_RANDOM_NUM = 1;
    private static final int MAX_RANDOM_NUM = 45;
    private static final int RANDOM_NUM_COUNT = 6;

    private RandomGenerator() {
    }

    public static List<Integer> generateRandomNum() {
        return List.copyOf(pickUniqueNumbersInRange(MIN_RANDOM_NUM, MAX_RANDOM_NUM, RANDOM_NUM_COUNT));
    }
}

File: src/main/kotlin/lotto/utils/RandomGenerator.kt
package lotto.utils

import camp.nextstep.edu.missionutils.Randoms.pickUniqueNumbersInRange

object RandomGenerator {

    private const val MIN_RANDOM_NUM: Int = 1;
    private const val MAX_RANDOM_NUM: Int = 45;
    private const val RANDOM_NUM_COUNT: Int = 6;

    fun generateRandomNum(): List<Int> =
        pickUniqueNumbersInRange(MIN_RANDOM_NUM, MAX_RANDOM_NUM, RANDOM_NUM_COUNT).toList()
}
